src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -163,6 +167,33 @@ public String toString() {
         return groups.asUnmodifiableObservableMap();
     }
 
+    @Override
+    public ObservableList<PersonEvent> getUpcomingDates() {
+        List<PersonEvent> personEvents = new ArrayList<>();
+        persons.forEach(person -> {
+            // Add birthday
+            personEvents.add(new PersonEvent(person.getBirthday().getBirthday(), person,
+                    PersonEvent.getBirthdayDescription(person)));
+            // Add special dates
+            person.getDates().forEach(event -> personEvents.add(new PersonEvent(event.getDate(), person,
+                    PersonEvent.getEventDescription(person, event))));
+        });
+
+        LocalDate now = LocalDate.now();
+        personEvents.sort((x, y) -> {
```

Comment:
I think this could be better.
```java
LocalDate xRelative = x.compareTo(now) < 0 ? x.plusYears(1) : x;
LocalDate yRelative = y.compareTo(now) < 0 ? y.plusYears(1) : y;
return xRelative.compareTo(yRelative);
```

Reply:
We can't simply compare `LocalDate`s as we're not taking year into account here, since the events are repeated annually.

Reply:
👍